app/services/targets/create_service.rb

```diff
@@ -34,10 +34,15 @@ def create_match(target, compatible_target)
       [target, compatible_target].each do |target_to_update|
         target_to_update.update!(matched: true)
       end
+      notification(target.user)
     end
 
     def compatible_target(target)
       TargetQuery.new.compatible_target(target)
     end
+
+    def notification(user)
+      MatchNotificationMailer.send_match_email(user).deliver
```

Comment:
here you need to call a Job, then the Job should call the NotificationService, and the Service should call the MatchNotificationMailer